netrd/distance/graph_diffusion_distance.py

```diff
@@ -0,0 +1,156 @@
+"""
+gdd.py
+--------------------------
+
+Graph diffusion distance, from
+
+Hammond, D. K., Gur, Y., & Johnson, C. R. (2013, December). Graph diffusion
+distance: A difference measure for weighted graphs based on the graph Laplacian
+exponential kernel. In Global Conference on Signal and Information Processing,
+2013 IEEE (pp 419-422). IEEE. https://doi.org/10.1109/GlobalSIP.2013.6736904
+
+author: Brennan Klein
+email: brennanjamesklein at gmail dot com
+Submitted as part of the 2019 NetSI Collabathon.
+
+"""
+
+import numpy as np
+import networkx as nx
+from scipy.sparse.csgraph import laplacian
+from .base import BaseDistance
+from ..utilities import undirected
+
+
+class GraphDiffusion(BaseDistance):
+    """Find the maximally dissimilar diffusion kernels between two graphs."""
+
+    @undirected
+    def dist(self, G1, G2, thresh=1e-14, resolution=1000):
```

Comment:
I'd use `thresh=1e-8` or something like that. Eigensolvers rarely use machine precision by default.

Reply:
Switched to `1e-8`.